src/store/index.js

```diff
@@ -160,23 +160,21 @@ export const store = new Vuex.Store({
                 } */
             });
         },
-        getNeedData() {
-            // TO DO: Move to UserData
-            firebase.firestore().collection('needs/2/social').orderBy("timeStamp", "desc").limit(1).get().then((querySnapshot)=>{
+        getUserData() {
+            // Get start date of the week
+            let today = moment();
+            let from_date = today.startOf('isoWeek').format();
+
+            // Get Social entry of the day
+            firebase.firestore().collection('needs/2/social').where("timeStamp", ">", from_date).orderBy("timeStamp", "desc").get().then((querySnapshot)=>{
```

Comment:
Tip: you can add prettier (https://www.npmjs.com/package/prettier) to the project and it'll help with formatting.

This line and a few lines below are too long, you can split it at the dots.
e.g.

firebase.firestore()
  .collection('needs/2/social')
  .where("timeStamp", ">", from_date)
  .orderBy("timeStamp", "desc")
  .get()
  .then((querySnapshot)=>{